lib/agriculture_sensor_parser.ex

```diff
@@ -0,0 +1,354 @@
+        parsed_payload = Map.put_new(parsed_payload, :configuration_control_commands, "There is unparsed data left! (#{byte_size(x)} Bytes) => (#{Base.encode16(x)})")  # To see in Control/Field
+        
+        [parsed_payload]
+        
+      # Default behaviour -> no bytes left
+      _ -> 
+        [parsed_payload] # Returns the payload.. could be empty
+    end
+  end
+  
+  
+# ---
+
+  
+  # Definition of Fields
+  def fields() do
+    [
+      %{
+        "field"   =>"battery_remaining_lifetime",
+        "unit"    =>"%",
+        "display" =>"Batterie (restliche Lebenszeit)"
+      },
+      %{
+        "field"   =>"input_1",
+        "unit"    =>"kHz", # no Conversion factor provided
+        "display" =>"Bodenfeuchtigkeit"
+      },
+      %{
+        "field"   =>"input_2",
+        "unit"    =>"mV", # no Conversion factor provided
+        "display" =>"Bodentemperatur"
+      },
+      %{
+        "field"   =>"watermark_01",
+        "unit"    =>"Hz", # no Conversion factor provided
+        "display" =>"Bodenwasserspannung 1"
+      },
+      %{
+        "field"   =>"watermark_02",
+        "unit"    =>"Hz", # no Conversion factor provided
+        "display" =>"Bodenwasserspannung 2"
+      },
+      %{
+        "field"   =>"ambient_light_intensity",
+        "unit"    =>"lx",
+        "display" =>"Umgebungslichtintensität"
+      },
+      %{
+        "field"   =>"ambient_light_alarm",
+        "display" =>"Umgebungslichtalarm"
+      },
+      %{
+        "field"   =>"accelerometer_data",
+        "unit"    =>"milli-𝑔",
+        "display" =>"Beschleunigungsmesserdaten"
+      },
+      %{
+        "field"   =>"impact_magnitude",
+        "unit"    =>"milli-𝑔",
+        "display" =>"Aufprallgröße"
+      },
+      %{
+        "field"   =>"impact_alarm",
+        "display" =>"Aufprallalarm"
+      },
+      %{
+        "field"   =>"ambient_temperature",
+        "unit"    =>"°C",
+        "display" =>"Umgebungstemperatur"
+      },
+      %{
+        "field"   =>"ambient_relative_humidity",
+        "unit"    =>"%",
+        "display" =>"relative Luftfeuchtigkeit"
+      },
+      %{
+        "field"   =>"mcu_temperature",
+        "unit"    =>"°C",
+        "display" =>"MCU Temperatur"
+      }
+    ]
+  end
+  
+  
+  # TEST's
+  def tests() do
+    # Note: please reorder/cut/add the payload to test dynamic parsing behaviour (in a new element down here 👇)
+    [
+      { # All in one (ordered by documentation listing)
+        :parse_hex, "00FF0001040000020200000504000006040000096500000900000A710000000000000A0200000A00000B6700000B68000C6700", %{meta: %{frame_port: 10}}, %{
+          battery_remaining_lifetime: 0,
+          input_1: 0,
+          input_2: 0,
+          watermark_01: 0,
+          watermark_02: 0,
+          ambient_light_intensity: 0,
+          ambient_light_alarm: false,
+          accelerometer_data: %{x: 0, y: 0, z: 0},
```

Comment:
No sub-structures possible. Use `accelerometer_x` ... instead.

Reply:
Ok, I hoped it would at least render to an JSON-Array. Is there a reason for the use of a plain data format?